common/src/main/scala/com/gu/media/upload/actions/UploadActionHandler.scala

```diff
@@ -140,6 +148,25 @@ abstract class UploadActionHandler(store: UploadsDataStore, plutoStore: PlutoDat
     }
   }
 
+  private def transcodeToS3(fileName: String, pipelineId: String) {
+
+    val jobInput = new JobInput().withKey(fileName)
+
+    val mp4FileName = fileName + ".mp4"
```

Comment:
I think mp4 is ok initially, especially as self hosting will be a rare occurrence.

I've got a secret desire to produce a single [DASH encoding](https://en.wikipedia.org/wiki/Dynamic_Adaptive_Streaming_over_HTTP) so all clients get adaptive playback... but that's a long way away!

Although m3u8s are adaptive too, support on Android isn't great.